src/state_db.rs

```diff
@@ -49,6 +57,30 @@ pub struct StateDB {
     options: DBOptions,
 }
 
+impl<'a> CurrentState<'a> {
+    fn new(root: &'a [u8], version: BlockHeight) -> Self {
+        Self { root, version }
+    }
+
+    fn to_bytes(&self) -> Vec<u8> {
+        [self.root, &self.version.to_be_bytes()].concat()
+    }
+
+    fn from_bytes(bytes: &'a [u8]) -> Self {
+        if bytes.len() < 4 {
```

Comment:
why is it 4 here root has 32 bytes?
maybe in this case it could be error?

Reply:
Yes, this not necessary, I removed that checking.